src/jit/lowerxarch.cpp

```diff
@@ -417,6 +417,9 @@ void Lowering::TreeNodeInfoInit(GenTree* tree)
 
         case GT_MUL:
         case GT_MULHI:
+#if defined(_TARGET_X86_) && !defined(LEGACY_BACKEND)
+        case GT_MUL_HI:
```

Comment:
This entire file is already under !defined(LEGACY_BACKEND)

This comment applies to other #ifdefs you've added here as well.